File: mobile/lib/services/ai_chat_service.dart
import '../config/api_client.dart';

class AiChatService {
  final _apiClient = ApiClient();

  Future<Map<String, dynamic>> sendMessage({
    required String message,
    required String language,
    String? conversationId,
  }) async {
    final response = await _apiClient.dio.post('/ai-chat/message', data: {
      'message': message,
      'language': language,
      if (conversationId != null) 'conversationId': conversationId,
    });
    return Map<String, dynamic>.from(response.data);
  }
}

File: web/src/services/aiChat.service.ts
import { apiClient } from '../config/api';

export type ChatLanguage = 'en' | 'te' | 'hi';

export interface ChatMessageResponse {
  conversationId: string;
  messageId: string;
  content: string;
  requiresEscalation: boolean;
  propertySuggestions?: Array<{
    propertyId: string;
    title: string;
    price: number;
    location: string;
    matchScore: number;
    matchReasons: string[];
  }>;
}

export const aiChatService = {
  async sendMessage(message: string, language: ChatLanguage, conversationId?: string) {
    const response = await apiClient.post('/ai-chat/message', {
      message,
      language,
      conversationId,
    });
    return response.data as ChatMessageResponse;
  },
};
